src/components/Form/InputWrapper.tsx

```diff
@@ -0,0 +1,34 @@
+import type {ForwardedRef} from 'react';
+import React, {forwardRef, useContext} from 'react';
+import type AmountTextInput from '@components/AmountTextInput';
+import type CheckboxWithLabel from '@components/CheckboxWithLabel';
+import type Picker from '@components/Picker';
+import type SingleChoiceQuestion from '@components/SingleChoiceQuestion';
+import TextInput from '@components/TextInput';
+import type {BaseTextInputRef} from '@components/TextInput/BaseTextInput/types';
+import FormContext from './FormContext';
+import type {BaseInputProps, InputWrapperProps} from './types';
+
+// TODO: Add remaining inputs here once these components are migrated to Typescript:
+// AddressSearch | CountrySelector | StatePicker | DatePicker | EmojiPickerButtonDropdown | RoomNameInput | ValuePicker
+type ValidInputs = typeof TextInput | typeof AmountTextInput | typeof SingleChoiceQuestion | typeof CheckboxWithLabel | typeof Picker;
+
+function InputWrapper<TInput extends ValidInputs, TInputProps extends BaseInputProps>(
+    {InputComponent, inputID, valueType = 'string', ...rest}: InputWrapperProps<TInput, TInputProps>,
+    ref: ForwardedRef<BaseTextInputRef>,
+) {
+    const {registerInput} = useContext(FormContext);
+    // There are inputs that don't have onBlur methods, to simulate the behavior of onBlur in e.g. checkbox, we had to
+    // use different methods like onPress. This introduced a problem that inputs that have the onBlur method were
+    // calling some methods too early or twice, so we had to add this check to prevent that side effect.
+    // For now this side effect happened only in `TextInput` components.
+    const shouldSetTouchedOnBlurOnly = InputComponent === TextInput;
+
+    // TODO: Sometimes we return too many props with register input, so we need to consider if it's better to make the returned type more general and disregard the issue, or we would like to omit the unused props somehow.
+    // eslint-disable-next-line react/jsx-props-no-spreading, @typescript-eslint/no-explicit-any
+    return <InputComponent {...(registerInput(inputID, {ref, valueType, ...rest, shouldSetTouchedOnBlurOnly}) as any)} />;
```

Comment:
Maybe we can update `BaseInputProps` to be a union of components props (TextInput, Picker, etc)

Or I guess it should be possible to have a common `BaseInputProps` type with common fields reused in every input component typing, maybe it can help too

Reply:
> Maybe we can update BaseInputProps to be a union of components props (TextInput, Picker, etc)

This was what we initially coded, but we faced some generic typing problems 😅 So we end up with this 

> Or I guess it should be possible to have a common BaseInputProps type with common fields reused in every input component typing, maybe it can help too

This is what we have now, right?

Reply:
But `BaseInputProps` type is not used in components typing, so it's not used in `BasePickerProps` or `BaseTextInputProps` types for example. Maybe it's something that can be done in the future when all components will be migrated

Reply:
I agree, that should be easier once all components are in TS

Reply:
I'm running into issues with this...

> Maybe we can update BaseInputProps to be a union of components props (TextInput, Picker, etc)

Better yet, can we detect the props for the specific type being used by the `InputWrapper`? So if I'm using a `TextInput` prop, I know I can pass all the props from `InputWrapper` itself, or any of the props from `TextInput`, but none of the props from `Picker`.

Reply:
> Better yet, can we detect the props for the specific type being used by the InputWrapper? So if I'm using a TextInput prop, I know I can pass all the props from InputWrapper itself, or any of the props from TextInput, but none of the props from Picker.

[I think this is what we are doing at the moment](https://github.com/Expensify/App/blob/main/src/components/Form/types.ts#L45-L49), if I pass `SingleChoiceQuestion` to `InputWrapper`'s `InputComponent` prop, `InputWrapper` will require `SingleChoiceQuestion` props like `currentQuestionIndex`, `possibleAnswers`, etc.